huaweicloud/services/cc/resource_huaweicloud_cc_central_network_policy_apply.go

```diff
@@ -0,0 +1,293 @@
+
+			targetStatus := []string{
+				"true",
+			}
+			if utils.StrSliceContains(targetStatus, status) {
+				return applyPolicyRespBody, "COMPLETED", nil
+			}
+
+			pendingStatus := []string{
+				"false",
+			}
+			if utils.StrSliceContains(pendingStatus, status) {
+				return applyPolicyRespBody, "PENDING", nil
+			}
+
+			return applyPolicyRespBody, status, nil
+		},
+		Timeout:      t,
+		Delay:        10 * time.Second,
+		PollInterval: 10 * time.Second,
+	}
+	_, err := stateConf.WaitForStateContext(ctx)
+	return err
+}
+
+func resourceCentralNetworkPolicyApplyRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	var mErr *multierror.Error
+
+	var (
+		getCentralNetworkPolicyApplyHttpUrl = "v3/{domain_id}/gcn/central-network/{central_network_id}/policies"
+		getCentralNetworkPolicyApplyProduct = "cc"
+	)
+	getCentralNetworkPolicyApplyClient, err := cfg.NewServiceClient(getCentralNetworkPolicyApplyProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating CC client: %s", err)
+	}
+
+	getCentralNetworkPolicyApplyPath := getCentralNetworkPolicyApplyClient.Endpoint + getCentralNetworkPolicyApplyHttpUrl
+	getCentralNetworkPolicyApplyPath = strings.ReplaceAll(getCentralNetworkPolicyApplyPath, "{domain_id}", cfg.DomainID)
+	getCentralNetworkPolicyApplyPath = strings.ReplaceAll(getCentralNetworkPolicyApplyPath, "{central_network_id}", d.Id())
+
+	getCentralNetworkPolicyApplyOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			200,
+		},
+		MoreHeaders: map[string]string{"Content-Type": "application/json"},
+	}
+
+	getCentralNetworkPolicyApplyResp, err := getCentralNetworkPolicyApplyClient.Request("GET", getCentralNetworkPolicyApplyPath,
+		&getCentralNetworkPolicyApplyOpt)
+
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error retrieving central network policy")
+	}
+
+	getCentralNetworkPolicyApplyRespBody, err := utils.FlattenResponse(getCentralNetworkPolicyApplyResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	getCentralNetworkPolicyApplyRespBody = utils.PathSearch("central_network_policies[?is_applied]|[0]",
+		getCentralNetworkPolicyApplyRespBody, nil)
+	if getCentralNetworkPolicyApplyRespBody == nil {
+		return common.CheckDeletedDiag(d, golangsdk.ErrDefault404{}, "no data found")
+	}
+
+	mErr = multierror.Append(
+		mErr,
+		d.Set("region", region),
+		d.Set("central_network_id", utils.PathSearch("central_network_id", getCentralNetworkPolicyApplyRespBody, nil)),
+		d.Set("policy_id", utils.PathSearch("id", getCentralNetworkPolicyApplyRespBody, nil)),
+	)
+
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func resourceCentralNetworkPolicyApplyDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	// get the default policy id which is not associated with the er instance and the version is 1.
```

Comment:
Is it always true?
how about add a new parameter `apply` in huaweicloud_cc_central_network_policy?

Reply:
yes, it's always true;
we can create many policies, but only one can be applied.  
If the `apply`  field of both policies is true, you will not know which one is applied.